rosmon_core/src/main.cpp

```diff
@@ -356,13 +357,13 @@ int main(int argc, char** argv)
                         {
                                 logDir.pop_back();
                         }
-                        logDir = logDir + "/rosmon";
+                        workDir = logDir = logDir + "/rosmon";
                         if (chdir(logDir.c_str()) == 0 || mkdir(logDir.c_str(), 0777) == 0) 
                         {
-                                std::string dir = logDir + "/roslogs";
-                                if (chdir(dir.c_str()) == 0 || mkdir(dir.c_str(), 0777) == 0) 
+                                logDir = logDir + "/roslogs";
+                                if (chdir(logDir.c_str()) == 0 || mkdir(logDir.c_str(), 0777) == 0) 
                                 {
-                                        logFile = dir + "/" + launchInfo.launch_group + "_" + launchInfo.launch_config + "_" + launchInfo.robot_name + ".log";
+					logFile = logDir + "/" + launchInfo.launch_group + "_" + launchInfo.launch_config + ".log";
```

Comment:
we can drop this now, since you are writing to each node

Reply:
This log still provides information. If a node dies, it is written in this log.